apps/frontend/src/components/ResponseBox/ResponseBox.tsx

```diff
@@ -1,28 +1,87 @@
+import { acceptResponse, deleteResponse } from "../../services/responses";
 import { useRevalidator } from 'react-router';
 import CustomBtn from '../CustomBtn/CustomBtn';
 
 type Props = {
-  response: ResponseWithUser;
+  response: ResponseWithUserAndRequest;
+  requestOwnerId: number;
 }
 
-export default function ResponseBox({ response }: Props) {
+function isLoggedUserRequestOwner(userId: number, requestOwnerId: number) {
+  return userId === requestOwnerId;
+};
+
+function isLoggedUserResponseOwner(userId: number, responseOwnerId: number) {
+  return userId === responseOwnerId;
+}
+
+export default function ResponseBox({ response, requestOwnerId }: Props) {
   const revalidator = useRevalidator();
 
+  let loggedUser = localStorage.getItem('user');
+  let loggedUserId = loggedUser ? JSON.parse(loggedUser).id : null;
+
   const date = String(response.time_created).split("T")[0];
+
   function handleAcceptOffer() {
     acceptResponse(String(response.id));
     revalidator.revalidate();
   }
 
-  const contactInfo = response.status === "ACCEPTED" ? (
-    <div>
-      <p className={styles.p}>You've accepted this offer for help.</p>
-      <p className={styles.p}>Contact at: <span>{response.user.email}</span></p>
-    </div>
-  ) : <CustomBtn variant='outline-dark' className={styles.btn} onClick={handleAcceptOffer}>Accept Offer</CustomBtn>;
+  function handleDeleteResponse() {
+    deleteResponse(String(response.id));
+    revalidator.revalidate();
+  }
+
+  /*
+    Cases:
+    1. User is not the response or request owner;
+      Show nothing;
+    2. User is also response owner:
+      Show a delete and edit button;
+      - if response status is ACCEPTED also show msg "You're response has been accepted"
+    3. User is also request owner:
+      Show a 'Accept offer' button;
+      - if response status is ACCEPTED show contact info.
+  */
+
+  function displayContactInfo() {
+    const requestOwner = isLoggedUserRequestOwner(loggedUserId, requestOwnerId);
+    const responseOwner = isLoggedUserResponseOwner(loggedUserId, response.user_id);
+
+    if (!(requestOwner || responseOwner)) return null;
+      
+    if (requestOwner) {
+      if (response.status === "ACCEPTED") {
+        return (
+          <>
+            <p className={styles.p}>You've accepted this offer for help.</p>
+            <p className={styles.p}>Contact at: <span>{response.user.email}</span></p>
+          </>
+        )
+      } else {
+        return (
+          <CustomBtn variant='outline-dark' className={styles.btn} onClick={handleAcceptOffer}>Accept Offer</CustomBtn>
+        )
+      }
+    } else if (responseOwner) {
+      if (response.status === "ACCEPTED") {
+        return (
+          <>
+            <p className={styles.p}>Your help offer has been accepted.</p>
+            <CustomBtn variant='danger' className={`${styles.btn} ${styles.deleteResponseBtn}`} onClick={handleDeleteResponse}>Delete Response</CustomBtn>
```

Comment:
Are you also planning to add the Edit button here?

Reply:
Yep!